internal/fields/validate.go

```diff
@@ -320,6 +320,11 @@ func (v *Validator) validateDocumentValues(body common.MapStr) multierror.Error
 				continue
 			}
 
+			// skip checking `data_stream.dataset` because reroute processor will change this field value
+			if datasetField == "data_stream.dataset" {
```

Comment:
Rather than skipping the validation, would it make sense to set up the validator with the new expected dataset instead? Disclaimer that I'm not familiar with the code base.